src/common/registry/fetch/utils/batches.ts

```diff
@@ -0,0 +1,13 @@
+export const makeBatches = (batchSize: number, offset: number, totalAmount: number) => {
+  if (batchSize < 1) throw new Error('batchSize must be greater than 0');
+  const numberOfBatches = Math.ceil(totalAmount / batchSize);
+  const batches: { offset: number; batchSize: number }[] = [];
+
+  for (let i = 0; i < numberOfBatches; i++) {
+    const currentOffset = offset + i * batchSize;
+    const currentBatchSize = Math.min(batchSize, totalAmount - i * batchSize);
```

Comment:
maybe better const currentBatchSize = Math.min(batchSize, totalAmount - offset - i * batchSize);

Reply:
more correct

Reply:
The same as above, `totalAmount` is actually a `limit`